pkg/cloudprovider/provider/vsphere/provider.go

```diff
@@ -88,7 +92,73 @@ func (vsphereServer VSphereServer) Status() instance.Status {
 }
 
 func (p *provider) AddDefaults(spec v1alpha1.MachineSpec) (v1alpha1.MachineSpec, bool, error) {
-	return spec, false, nil
+	changed := false
+
+	cfg, _, rawCfg, err := p.getConfig(spec.ProviderConfig)
+	if err != nil {
+		return spec, changed, cloudprovidererrors.TerminalError{
+			Reason:  v1alpha1.InvalidConfigurationMachineError,
+			Message: fmt.Sprintf("Failed to parse MachineSpec, due to %v", err),
+		}
+	}
+
+	// default templatenetname to first network of template if none specific was given.
+	if cfg.TemplateNetName == "" {
```

Comment:
Please change the condition here to `if cfg.TemplateNetName == "" && config.VMNetName != ""` so we only act if is actually required